backend/app/domains/courses/crud.py

```
@@ -35,7 +46,29 @@ def get_course(db: Session, course_id: int) -> Course | None:
     )
     return db.exec(stmt).all()
+
+
+def delete_course(db: Session, course: Course, current_user: User):
+    deleted_registrations = delete_registrations_by_course_id(db, course.id, current_user)
+    db.delete(course)
+    db.commit()
+    return None
+
+
+def get_courses_by_professor(db: Session, current_user: User) -> List[CourseRead]:
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)